File: 김현경/[LV3] 섬 연결하기.py
def find(parent, x):
    if parent[x] != x:
        parent[x] = find(parent, parent[x])
    return parent[x]
    
def union(parent, a, b):
    a = find(parent, a)
    b = find(parent, b)
    if a < b:
        parent[b] = a
    else:
        parent[a] = b
    
def solution(n, costs):
    total_cost = 0
    parent = [i for i in range(n)]
    costs.sort(key=lambda x: x[2])
    
    for cost in costs:
        a, b, c = cost
        if find(parent, a) != find(parent, b):
            union(parent, a, b)
            total_cost += c
    return total_cost

File: 조성원/[LV3] 섬 연결하기.js
function solution(n, costs) {
  const find = (x) => {
    if (cycleTable[x] === x) return x;
    cycleTable[x] = find(cycleTable[x]);
    return cycleTable[x];
  };

  const union = (rootA, rootB) => {
    if (rootA < rootB) cycleTable[rootB] = rootA;
    else cycleTable[rootA] = rootB;
  };

  let answer = 0;

  costs.sort((a, b) => a[2] - b[2]);

  const cycleTable = Array.from({ length: n }, (_, i) => i);

  for (const [a, b, cost] of costs) {
    const rootA = find(a);
    const rootB = find(b);

    if (rootA !== rootB) {
      union(rootA, rootB);
      answer += cost;
    }
  }

  return answer;
}
